windows/hid.c

```diff
@@ -318,11 +324,17 @@ static void register_string_error_to_buffer(wchar_t **error_buffer, const WCHAR
 
 static void register_winapi_error(hid_device *dev, const WCHAR *op)
 {
+	if (!dev)
```

Comment:
functions like this are used only internally
is it even possible accidentally pass an invalid `dev` here? I'm pretty sure if the user passes an invalid or NULL `dev` into any of the `hid_*` functions - there woud be an undefined behavior (crash) way before we get here in most places...

Reply:
I don't know, I didn't discriminate when I added these.
In my opinion there shouldn't be any function without sanity checks.
It's not unthinkable that this was possible before I added the other checks after all.

Reply:
> In my opinion there shouldn't be any function without sanity checks.

That is actually debatable. Most of the checks in this PR are actually do more harm then help.
See my other comment below.